src/Rarity/Rarity.router.ts

```diff
@@ -1,18 +1,101 @@
 import { server } from 'decentraland-server'
-
+import { Request } from 'express'
 import { Router } from '../common/Router'
 import { collectionAPI } from '../ethereum/api/collection'
+import { RarityFragment } from '../ethereum/api/fragments'
+import { HTTPError, STATUS_CODES } from '../common/HTTPError'
+import { isUsingRaritiesWithOracle, getRarityFromBlockchain } from './utils'
+import { Currency, Rarity } from './types'
 
 export class RarityRouter extends Router {
   mount() {
-    /**
-     * Returns the available rarities
-     */
+    // Returns the available rarities.
     this.router.get('/rarities', server.handleRequest(this.getRarities))
+
+    // Returns a single rarity according to the rarity name provided.
+    this.router.get('/rarities/:name', server.handleRequest(this.getRarity))
+  }
+
+  getRarities = async (): Promise<Rarity[]> => {
+    const graphRarities = await collectionAPI.fetchRarities()
+
+    // If the server is still using the old rarities contract,
+    // return rarities as they have been always returned.
+    if (!isUsingRaritiesWithOracle()) {
+      return graphRarities
+    }
+
+    // Query the blockchain to obtain rarities with MANA prices converted from USD.
+    const blockchainRarities = await Promise.all(
+      graphRarities.map((rarity) => this.getRarityFromBlockchain(rarity.id))
+    )
+
+    // Convert the array into a Map for an easier lookup
+    const blockchainRaritiesMap = new Map(
+      blockchainRarities.map((rarity) => [rarity.name, rarity])
+    )
+
+    // Consolidate rarities obtained from the graph with rarities obtained
+    // from the blockchain.
+    return graphRarities.map((graphRarity) => {
+      // Not handling if the blockchain rarity is not present because that is checked when calling the
+      // this.getRarityFromBlockchain function.
+      const blockchainRarity = blockchainRaritiesMap.get(graphRarity.name)!
+
+      return {
+        ...graphRarity,
+        price: blockchainRarity.price,
+        originalPrice: graphRarity.price,
+        originalCurrency: Currency.USD,
```

Comment:
As "Original" as name might be a bit confusing, what if we change this to a map?

As a map:
```
price: {
'MANA': '4000000000000000000',
'USD': '10000000000000000000'
}
```